### ISSUE TITLE
Is it possible to fetch system messages?

### ISSUE THREAD

**Issue by kiddick** [tvdstaaij/telegram-history-dump#49]
I mean like `user has joined` or `someone added someone` (to chat) and so on. This is kind of faceless messages though I can see them in desktop app.

**Comment by tvdstaaij** [comment]
Yes, these are called "service" messages, and can be identified by a `service` field set to `true` in the JSON files. So the raw JSON data contains all of them, and most builtin formatters handle at least a few of these ([example](https://github.com/tvdstaaij/telegram-history-dump/blob/9c90d68a6d6da07516a12058daf470aaf07abb06/formatters/plaintext.rb#L44-L54)).

Now that I think of it, I didn't even consider a "user has joined" scenario back then, only "user A was added by user B" — that was before public groups were a thing. Perhaps this would be a good time for me to take a look at message formatting again, since quite a few things have changed over time.

**Comment by kiddick** [comment]
Seems service messages which in group chat before migration to supergroup are losing.

**Comment by kiddick** [comment]
Hmm.. I can't even find common messages in history which were before migration to supergroup..

**Comment by tvdstaaij** [comment]
Yes, Telegram considers the supergroup as a new entity with its own history, so when converting to a supergroup the history is not inherited. The old group will still exist though and can be dumped separately.

I do not think telegram-cli provides solid information that links the group and supergroup together, it's just that the old one will end with a migrated_to event and the new supergroup will start with a migrated_from event. If telegram-cli were to provide such information then perhaps it would be possible to merge them together during processing.
